File: test/ayush.dev/scala/pipelines/scala_external_pipeline/code/src/main/scala/io/prophecy/pipelines/scala_external_git_pipeline/graph/Reformat_1.scala
package io.prophecy.pipelines.scala_external_git_pipeline.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_external_git_pipeline.config.ConfigStore._
import io.prophecy.pipelines.scala_external_git_pipeline.udfs.UDFs._
import io.prophecy.pipelines.scala_external_git_pipeline.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import java.time._

object Reformat_1 {

  def apply(spark: SparkSession, in: DataFrame): DataFrame =
    in.select(
      col("customer_id"),
      col("first_name"),
      col("last_name"),
      col("phone"),
      col("email"),
      col("country_code"),
      col("account_open_date"),
      col("account_flags"),
      concat(lit(Config.c_string),
             col("customer_id"),
             intudf(col("customer_id").cast(IntegerType))
      ).as("c_config")
    )

}

File: test/ayush.dev/python/pipelines/PythonInsideProjectDatasetsOnly/code/pythoninsideprojectdatasetsonly/graph/Reformat_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from pythoninsideprojectdatasetsonly.config.ConfigStore import *
from pythoninsideprojectdatasetsonly.udfs.UDFs import *

def Reformat_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        col("customer_id"), 
        col("first_name"), 
        col("last_name"), 
        col("phone"), 
        col("email"), 
        col("country_code"), 
        col("account_open_date"), 
        col("account_flags"), 
        lookup("LookupMain", col("customer_id"), col("email")).getField("first_name").alias("c_lookup"), 
        concat(lit(Config.c_string), col("first_name"), squared(col("customer_id").cast(IntegerType())))\
          .alias("c_config")
    )
